metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/repositories/SampleRepositoryCustom.java

```diff
@@ -39,4 +45,29 @@ default Sample findOrSave(Sample sample) {
         }
         return save(sample);
     }
+
+    default List<Sample> findOrSaveList(List<Sample> sampleIn) {
+        List<Sample> existingSampleList;
+        Predicate predicate = qSample.accessionVersionId.accession.in(sampleIn.stream()
```

Comment:
This query is wrong, it would match A1V2 and A2V1 as well in the example eg:{A1V1, A2V2}